p2p/proto/consensus.proto

```diff
@@ -53,27 +53,29 @@ message ProposalInit {
 // The primary content of the block. This is opaque to the consensus module.
 message ProposalContent {
     oneof content {
-        Transactions  transactions        = 1;
-        BlockProof    proof               = 2;
-        // A list of precommits which supported the previous block. When `H+1` is decided this will
-        // become the canonical set of precommits for H which will determine the rewards.
-        //
-        // Note that the proof for `H` included in `H+K` or sent to L1 doesn't depend on this, as
-        // it merely commits us to the statement that "H had a quorum of precommits from Validators"
-        // without any commitment to the specific set of precommits used to prove that.
-        repeated Vote previous_precommits = 3;
+        // The timestamp field in metadata can impact consensus, specifically the lower bound
+        // applied. If nodes apply a lower bound validation based on their local time, then we
+        // risk a scenario where in round `R` proposal P is locked. Then in a later round the
+        // timestamp in P has gone stale. Therefore the lower bound should be "greater than the
+        // previous timestamp". Upper bounds don't suffer from this problem.
+        BlockMetadata metadata            = 1;
```

Comment:
Where is BlockMetadata defined? 

Also, it may be confusing if header.proto has messages that are different than here (e.g. BlockLocation is only here)

Reply:
It seems I hallucinated this, but let me describe my thoughts. A block is made of: state diff (built from TX list), proof, metadata (max cairo steps, gas price, height, parent hash, etc.). Basically we already have dedicated fields for TX batches and the proof, so I thought we needed another field for all of the other info that is sent.

Where should we define this? Do you have a different name you prefer?

Reply:
Discussed, no current usage so removed.